features/cellular/framework/targets/RiotMicro/AT/RM1000_AT_CellularStack.cpp

```diff
@@ -0,0 +1,412 @@
+{
+    return (protocol == NSAPI_UDP || protocol == NSAPI_TCP);
+}
+
+nsapi_error_t RM1000_AT_CellularStack::create_socket_impl(CellularSocket *socket)
+{
+    tr_debug("RM1000_AT_CellularStack::create_socket_impl");
+
+    int sock_id = SOCKET_UNUSED;
+
+    _at.lock();
+    if (socket->proto == NSAPI_UDP) {
+        _at.cmd_start("AT+RSOCR=0");
+        _at.cmd_stop();
+
+        _at.resp_start("+RSOCR:");
+        sock_id = _at.read_int();
+        _at.resp_stop();
+    } else if (socket->proto == NSAPI_TCP) {
+        _at.cmd_start("AT+RSOCR=1");
+        _at.cmd_stop();
+
+        _at.resp_start("+RSOCR:");
+        sock_id = _at.read_int();
+        _at.resp_stop();
+    } // Unsupported protocol is checked in "is_protocol_supported" function
+
+    if ((_at.get_last_error() != NSAPI_ERROR_OK) || (sock_id == -1)) {
+        _at.unlock();
+        tr_error("RM1000_AT_CellularStack::create_socket_impl error sock_id=%d err=%d", sock_id, _at.get_last_error());
+        return NSAPI_ERROR_NO_SOCKET;
+    }
+    _at.unlock();
+
+    // Check for duplicate socket id delivered by modem
+    for (int i = 0; i < RM1000_MAX_SOCKET; i++) {
+        CellularSocket *sock = _socket[i];
+        if (sock && sock != socket && sock->id == sock_id) {
+            return NSAPI_ERROR_NO_SOCKET;
+        }
+    }
+
+    socket->id = sock_id;
+
+    return NSAPI_ERROR_OK;
+}
+
+nsapi_error_t RM1000_AT_CellularStack::socket_connect(nsapi_socket_t handle, const SocketAddress &addr)
+{
+    tr_debug("RM1000_AT_CellularStack::socket_connect");
+
+    CellularSocket *socket = (CellularSocket *)handle;
+
+    if (socket) {
+        if (socket->id == SOCKET_UNUSED) {
+            nsapi_error_t err = create_socket_impl(socket);
+            if (err != NSAPI_ERROR_OK) {
+                return err;
+            }
+        }
+    } else {
+        return NSAPI_ERROR_DEVICE_ERROR;
+    }
+
+    _at.lock();
+    _at.cmd_start("AT+RSOCO=");
+    _at.write_int(socket->id);
+    _at.write_string(addr.get_ip_address(), false);
+    _at.write_int(addr.get_port());
+    _at.cmd_stop();
+    _at.resp_start("+RSOCO:");
+    int socket_id = _at.read_int();
+    _at.resp_stop();
+    _at.unlock();
+
+    if ((_at.get_last_error() == NSAPI_ERROR_OK) && (socket_id == socket->id)) {
+        socket->remoteAddress = addr;
+        socket->connected = true;
+        return NSAPI_ERROR_OK;
+    }
+
+    return NSAPI_ERROR_NO_CONNECTION;
+}
+
+nsapi_size_or_error_t RM1000_AT_CellularStack::socket_sendto_impl(CellularSocket *socket, const SocketAddress &address,
+                                                                  const void *data, nsapi_size_t size)
+{
+    tr_debug("RM1000_AT_CellularStack::socket_sendto_impl");
+
+    int sent_len = 0;
+    pollfh fhs;
+    fhs.fh = _at.get_file_handle();
+    fhs.events = POLLIN;
+
+    if (socket->proto == NSAPI_UDP) {
+        if (size > RM1000_MAX_PACKET_SIZE) {
```

Comment:
Should UDP have handling similar to TCP, or is UDP packet size really restricted to 1024?

Reply:
both udp and tcp have the same handling now